server/src/main/java/com/flipkart/varadhi/cluster/MessageExchange.java

```diff
@@ -0,0 +1,57 @@
+package com.flipkart.varadhi.cluster;
+
+import com.flipkart.varadhi.cluster.messages.ClusterMessage;
+import com.flipkart.varadhi.cluster.messages.ResponseMessage;
+import com.flipkart.varadhi.exceptions.NotImplementedException;
+import com.flipkart.varadhi.utils.JsonMapper;
+import io.vertx.core.eventbus.DeliveryOptions;
+import io.vertx.core.eventbus.EventBus;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * This is a subset of similar methods from vertx.EventBus for messages exchange between nodes.
+ *
+ * Supported methods are
+ * send(): dispatch message to one of handler registered at <routeName>.<apiName>.send, delivery can be tracked using the future.
+ * request(): dispatch message to one of handler registered at <routeName>.<apiName>.request and wait for a response.
+ * publish(): would dispatch message to all handler registered at <routeName>.<apiName>.publish
+ */
+@Slf4j
+public class MessageExchange {
+    EventBus vertxEventBus;
+    DeliveryOptions deliveryOptions;
+
+    public MessageExchange(EventBus vertxEventBus, DeliveryOptions deliveryOptions) {
+        this.vertxEventBus = vertxEventBus;
+        this.deliveryOptions = deliveryOptions;
+    }
+
+    public CompletableFuture<Void> send(String routeName, String apiName, ClusterMessage msg) {
+        CompletableFuture<Void> future = new CompletableFuture<>();
+        vertxEventBus.request(
+                getPath(routeName, apiName, RouteMethod.SEND), JsonMapper.jsonSerialize(msg), deliveryOptions, ar -> {
```

Comment:
so, here the path is vertx event bus "address". right?
So I still have concerns on this as the #addresses will increase with time. And the total #address to be managed by vertx / zk, will be proportional to api paths x node count.

I also have some doubts on the "addressing".

each web-server node will register the handler against the same path. so when the controller begins a request ("if" it begins), then the request can land on any web-server node. right? Just validating here.

Another concern is the ordering of events. If I send 2 events on 2 different "path", then I suppose there is no guarantee of the ordering between them. event_1 on path_1 is sent first and then event_2 on path_2 is sent without waiting for the response of event_1, then we probably cant assume anything about the ordering of these events. Dont think that this is an issue per say, but just pointing out.

Reply:
`will be proportional to api paths x node count.` : huh I may be wrong here. as the address dont have ip in them. (yet)

Reply:
you are correct, number of watches would be approx equal to node count * api, but that's still bounded. 
open to re-work the address based on how they span post one or two flow implemented e2e.

Event ordering. -- aligned, i think we shouldn't rely on events on two independent path being ordered.